### PR TITLE
I68 create documentation for backend api

### LINKED ISSUE #68: Create Documentation for backend API endpoints

### PR DESCRIPTION
**Pull request by ChantelleKerr**
## Change Summary
**[Briefly summarise the changes that you made. Just high-level stuff]**
- document backend
- testing

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [x] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

`cd documentation`
`mkdocs serve`

resolve #68 

# Related Issue

- Resolve #68

### CODE REVIEW

File: client/src/main.ts [hunk 1]
```diff
@@ -40,8 +38,4 @@ const vuetify = createVuetify({
   icons
 })
 
-app.use(createPinia())
-app.use(router)
-app.use(vuetify)
-
-app.mount('#app')
+createApp(App).use(createPinia()).use(router).use(vuetify).mount('#app')
```
Comment (nisarg-dave) [inline client/src/main.ts:41]: I still can't seem to see the Pinia store in my Vue dev tools. Are there any additional steps that I'm missing?

Comment (nisarg-dave) [reply client/src/main.ts:42]: Never mind, I can see it now after main was merged into this branch.

File: documentation/docs/backend/testing.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Testing
+
+Django automated tests have been made to test all our endpoints. Unit tests can be run from within the backend Docker container using the command `python manage.py test` or you can directly execute it into the docker container `docker exec -it csf_server python manage.py test`.
```
Comment (nisarg-dave) [inline documentation/docs/backend/testing.md:3]: Is this saying that the tests can be run by first entering the container using `docker exec -it csf_server` followed by `python manage.py test` or the tests can be directly run inside the container using `docker exec -it csf_server python manage.py test`?

**Review by nisarg-dave — COMMENTED** [review]
Good work on documenting the model attributes and views for each of the endpoints. I've left some minor comments. Also, I wasn't able to run `mkdocs serve` inside the documentation directory. I think mkdocs is a Python package that you might have installed globally on your laptop. Do you think we would need a Docker image for mkdocs like we had one for WAIS?

File: documentation/docs/backend/testing.md (hunk 2 above)
Comment (nisarg-dave) [reply documentation/docs/backend/testing.md:3]: Should this then just say the tests can be run inside the backend Docker container using `docker exec -it csf_server python manage.py test`?

**Comment by ChantelleKerr** [comment]
> Good work on documenting the model attributes and views for each of the endpoints. I've left some minor comments. Also, I wasn't able to run `mkdocs serve` inside the documentation directory. I think mkdocs is a Python package that you might have installed globally on your laptop. Do you think we would need a Docker image for mkdocs like we had one for WAIS?

Oh yes that could be possbile. I can create another issue for a docs docker container

File: documentation/docs/backend/testing.md (hunk 2 above)
Comment (ChantelleKerr) [reply documentation/docs/backend/testing.md:3]: I've given two options on how to run the testing.

Comment (nisarg-dave) [reply documentation/docs/backend/testing.md:3]: Ok no worries.

**Pull request merged**